src/client/pfs_new_storage.go

```diff
@@ -128,7 +129,35 @@ func (c APIClient) GetTarConditional(repoName string, commitID string, path stri
 		}
 	}
 	return nil
+}
 
+// ListFileV2 streams FileInfoNewStorage similar to ListFileF
+func (c APIClient) ListFileV2(repoName string, commitID string, path string, f func(fileInfo *pfs.FileInfoNewStorage) error) (retErr error) {
+	defer func() {
+		retErr = grpcutil.ScrubGRPC(retErr)
+	}()
+	ctx, cf := context.WithCancel(c.Ctx())
+	defer cf()
+	req := &pfs.ListFileRequest{
+		File: NewFile(repoName, commitID, path),
+		Full: true,
+	}
+	client, err := c.PfsAPIClient.ListFileV2(ctx, req)
+	if err != nil {
+		return err
+	}
+	for {
+		finfo, err := client.Recv()
+		if err == io.EOF {
```

Comment:
The way I tend to structure things like this is:
```
if err != nil {
  if err == io.EOF {
    break
  }
  return err
}
```
I think it is a little bit nicer from a readability standpoint because it groups the non-nil error path.